### PR TITLE
Result pane

### PR DESCRIPTION
**Pull request by pahumadad**
there has been implemented a "result pane" as a separate class from "mainwin".

Functionalities added:
- Create a result label when an "operation line" is created (in the same line)

Next steps:
- Update the result label always a new key is pressed on the operation line

Note: This branch does not be able to merge yet!

### CODE REVIEW

File: blackboard/gui/result_pane.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def add_label(self, txt_op, value):
+        if Gdk.keyval_name(value.keyval) == 'Return':
+            x = txt_op.get_cursor_locations()
+            height = x.strong.height
+            pos = x.strong.y + height
+            index = pos / height
+            lb = Gtk.Label(str(index))
+            self.lb_list.insert(index, lb)
+            if (len(self.lb_list) - 1) > index:
```
Comment (fortizc) [inline blackboard/gui/result_pane.py]: When this code is executed (return was pressed) the terminal show an error
```
Traceback (most recent call last):
  File "/Users/fortiz/dev/priv/blackboard/blackboard/gui/result_pane.py", line 27, in add_label
    self.lb_list.insert(index, lb)
TypeError: integer argument expected, got float
```

File: blackboard/gui/result_pane.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    def __init_lb_list(self):
+        lb = Gtk.Label(str(0))
+        self.lb_list = []
+        self.lb_list.append(lb)
+        self.attach(lb, 0, 0, 1, 1)
+        lb.show()
+
+    def add_label(self, txt_op, value):
+        if Gdk.keyval_name(value.keyval) == 'Return':
```
Comment (fortizc) [inline blackboard/gui/result_pane.py]: This is good but I think that is better if you use  the inverse comparison
```python
if Gdk.keyval_name(value.keyval) <> 'Return':
    return
< put the rest of your code here without an extra indentation :-) >
```

File: blackboard/gui/result_pane.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            x = txt_op.get_cursor_locations()
```
Comment (fortizc) [inline blackboard/gui/result_pane.py]: All code related to the position and index calculation should be moved to a specific method

File: blackboard/gui/result_pane.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+from gi.repository import Gtk, Gio, Gdk  # noqa
+
+
+class result_pane(Gtk.Grid):
+    def __init__(self):
+        Gtk.Grid.__init__(self)
+        self.__init_lb_list()
+
+    def __init_lb_list(self):
```
Comment (fortizc) [inline blackboard/gui/result_pane.py]: This method adds a label, but you already have a method call `add_label`...

File: blackboard/gui/result_pane.py (hunk 1 above)
Comment (pahumadad) [reply blackboard/gui/result_pane.py]: Fixed in commit 2b2f7d2c7fbb9311c9d3c6e103237bc08676d002

File: blackboard/gui/result_pane.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import gi
+gi.require_version('Gtk', '3.0')
+
+from gi.repository import Gtk, Gio, Gdk  # noqa
+
+
+class result_pane(Gtk.Grid):
+    def __init__(self):
+        Gtk.Grid.__init__(self)
```
Comment (fortizc) [inline blackboard/gui/result_pane.py]: In python 3.x the better way to call a method from the parent class is using `super()` the syntax is as follow
```python
super().__init__()
```
I know, this is must be fixed in the mainwin class too

File: blackboard/gui/result_pane.py (hunk 1 above)
Comment (fortizc) [reply blackboard/gui/result_pane.py]: Good 👍

File: blackboard/gui/result_pane.py (hunk 2 above)
Comment (pahumadad) [reply blackboard/gui/result_pane.py]: Fixed in commit d3bd51a2b6ae4e9d234e16c2c0e3ad36a6aa3ecf

File: blackboard/gui/result_pane.py (hunk 3 above)
Comment (pahumadad) [reply blackboard/gui/result_pane.py]: Yes!
I've moved this to a method call "capture_pressed_key". Now, this method calls "__add_label" method.
See commit d3bd51a2b6ae4e9d234e16c2c0e3ad36a6aa3ecf

File: blackboard/gui/result_pane.py (hunk 4 above)
Comment (pahumadad) [reply blackboard/gui/result_pane.py]: You're right!
I've changed the "add_label" method and I've eliminated the "__init_lb_list" method.
See commit d3bd51a2b6ae4e9d234e16c2c0e3ad36a6aa3ecf

File: blackboard/gui/result_pane.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    def __add_label(self, index):
+        lb = Gtk.Label(str(index))
+        self.lb_list.insert(index, lb)
+        if (len(self.lb_list) - 1) > index:
+            self.insert_row(index)
+        self.attach(lb, 0, index, 1, 1)
+        lb.show()
+
+    def capture_pressed_key(self, txt_op, value):
```
Comment (fortizc) [inline blackboard/gui/result_pane.py]: This doesn't make me sense
Why a method called `capture_pressed_key` calculate an index and add a label?
`capture_pressed_key` is very generic name maybe this method should be called `__add_label_at_cursor` and `__add_label` must be called `__add_label_at`. Another method called `__get_cursor_position`should be implemented with the `txt_op` as parameter and at the end, this method should return the index

Comment (pahumadad) [reply blackboard/gui/result_pane.py]: OK
The methods and their names have been restructured as  you suggest. Commit #002365b7c03678aa17e6356611b34cfd598c1e74

Comment (pahumadad) [reply blackboard/gui/result_pane.py]: Sorry!
Commit 002365b7c03678aa17e6356611b34cfd598c1e74

**Comment by fortizc** [comment]
Awesome commits! only just need to correct the call to parent constructor to be ready to merge this branch

File: blackboard/gui/result_pane.py (hunk 5 above)
Comment (pahumadad) [reply blackboard/gui/result_pane.py]: Fixed in commit ce5a394d6f1958159436e768d1500a342096fbdd

**Comment by pahumadad** [comment]
I've implemented the method called "update_label" to insert a "result" in a label with a certain "index" (label list).
See commit 5728497357ecc0d5f0975c2e06f8da8ca785edad

**Comment by fortizc** [comment]
Cool, it seems to be ready to merge!

File: blackboard/gui/result_pane.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        lb.show()
+
+    def __get_cursor_position(self, txt_op):
+        x = txt_op.get_cursor_locations()
+        height = x.strong.height
+        pos = x.strong.y
+        index = int(pos / height)
+        return index
+
+    def __get_pressed_key(self, value):
```
Comment (fortizc) [inline blackboard/gui/result_pane.py:28]: This is over engineering.

**Pull request merged**